src/collective/easyform/fields.py

```diff
@@ -26,15 +25,36 @@
 from zope.schema.interfaces import IField
 
 
+@implementer(IForm)
+class GenericFormWrapper(object):
+    def __init__(self, view):
+        self.__view__ = view
+
+    def __getattr__(self, item):
+        return getattr(self.__view__, item)
+
+
 @implementer(IValidator)
 @adapter(IEasyForm, Interface, IEasyFormForm, IField, Interface)
-class FieldExtenderValidator(z3c_validator.SimpleFieldValidator):
-
+class FieldExtenderValidator(object):
     """ z3c.form validator class for easyform fields in the default fieldset"""
 
+    def __init__(self, context, request, view, field, widget):
+        self.context = context
+        self.request = request
+        self.view = view
+        self.field = field
+        self.widget = widget
+
     def validate(self, value):
         """ Validate field by TValidator """
-        super(FieldExtenderValidator, self).validate(value)
+        # First see if there is a another validator adapter for the field which isn't us
+        view = GenericFormWrapper(self.view)
+        assert providedBy(view)(IForm)
+        assert not providedBy(view)(IEasyFormForm)
```

Comment:
Please do not use `assert` statements. They may get removed by code optimisers (maybe when making a `.pyo` file).

Also, it looks really funky to me. I had to go in with a pdb to see what this does. I think the next lines are equivalent to yours:

```
IForm.providedBy(view)
not IEasyFormForm.providedBy(view)
```

So that would become something like this:

```
if not IForm.providedBy(view):
    raise ValueError("Oops")
if IEasyFormForm.providedBy(view):
    raise ValueError("Oops")
```

But I guess they can actually be removed. They seem only for debugging.

Reply:
yes  they were. removed